src/App.tsx

```diff
@@ -1,24 +1,195 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import cn from 'classnames';
+import * as postService from './api/todos';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { FilterType } from './types/FilterType';
+import { Error } from './types/Error';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoError } from './components/TodoError';
+import { TodoItem } from './components/TodoItem';
 
-const USER_ID = 0;
+const USER_ID = 11260;
+
+const getVisibleTodos = (todos: Todo[], status: FilterType) => {
+  return todos
+    .filter(todo => {
+      switch (status) {
+        case FilterType.Completed:
+          return todo.completed;
+
+        case FilterType.Active:
+          return !todo.completed;
+
+        default:
+          return true;
+      }
+    });
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(Error.None);
+  const [status, setStatus] = useState(FilterType.All);
+  const [query, setQuery] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingIds, setDeletingIds] = useState<number[]>([]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrorMessage(Error.Load));
+  }, []);
+
+  const visibleTodos = useMemo(() => getVisibleTodos(todos, status),
+    [todos, status]);
+
+  const todoCount = useMemo(() => todos.filter(todo => !todo.completed).length,
+    [todos]);
+
+  const isCompletedTodos = useMemo(() => todos.some(todo => todo.completed),
+    [todos]);
+
+  const isNoTodo = useMemo(() => todos.length === 0,
+    [todos]);
```

Comment:
at least this useMemo redundant.
it is not expensive operation to check exist or not todos.